contracts/core-token-vesting/src/contract.rs

```diff
@@ -80,9 +84,179 @@ pub fn execute(
         ExecuteMsg::Claim { denoms, recipient } => {
             claim(deps, env, info, denoms, recipient)
         }
+        ExecuteMsg::CreateCampaign {
+            vesting_schedule,
+            campaign_name,
+            campaign_description,
+            managers,
+        } => create_campaign(
+            deps,
+            info,
+            vesting_schedule,
+            campaign_name,
+            campaign_description,
+            managers,
+        ),
+        ExecuteMsg::RewardUsers { requests } => {
+            reward_users(deps, env, info, requests)
+        }
+        ExecuteMsg::DeactivateCampaign {} => {
+            deactivate_campaign(deps, env, info)
+        }
+        ExecuteMsg::Withdraw { amount } => withdraw(deps, env, info, amount),
     }
 }
 
+/// Deactivate a campaign and withdraw all unallocated funds
+/// This will also withdraw all unallocated funds from the contract
+/// and send them to the campaign owner.
+fn deactivate_campaign(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+) -> Result<Response, ContractError> {
+    let mut campaign = CAMPAIGN
+        .load(deps.storage)
+        .map_err(|_| StdError::generic_err("Failed to load campaign data"))?;
+
+    if campaign.owner != info.sender
+        && !campaign.managers.contains(&info.sender.to_string())
```

Comment:
I would restrict this ability to only the campaign owner (i.e. the treasury multisig). The risk is too great if an individual is able to call this and have the tokens transfered to them.